src/core/bpf/allow_bind/maps.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#pragma once
+
+/* <linux/types.h> must precede <bpf/bpf_helpers.h> due to integer types
+ * in bpf helpers signatures.
+ */
+#include <linux/types.h>
+#include <bpf/bpf_helpers.h>
+#include <linux/bpf.h>
+
+/* Binding is allowed only to ports specified in allow_ports map.
+ * Ports are in host order.
+ * Ports are the same for both v6 and v4 hooks.
+ * max_entries is set from user space with bpf_map__resize
+ * helper.
+ */
+
+struct allow_ports_t {
```

Comment:
given you are not re-using this map definition type across multiple maps, I'd just inline it and not introduce unnecessary type (struct allow_ports_t).

Reply:
@anakryiko

There is a request to distinguish between IPv4 and IPv6, so this will be useful after all :)  https://github.com/systemd/systemd/pull/17655#discussion_r554484297

Reply:
I'd still not do it and would just declare maps with anonymous structs inline. But that's my preference (though typical in a lot of BPF applications I've seen).